docs/form-app-script.js

```diff
@@ -0,0 +1,120 @@
+// This is the code behind the Google Form that is the first-round version of campaign creation: https://goo.gl/forms/aAcBrU408Kv2jrEr2
+
+// TODO change these to prod URLs eventually
+SPONSOR_URL = 'http://jingle.staging.prx.tech/api/v1/sponsors';
+PODCAST_URL = 'http://jingle.staging.prx.tech/api/v1/shows';
+CAMPAIGN_URL = 'http://jingle.staging.prx.tech/api/v1/campaigns';
+
+
+function onSubmit(e){
+  var attributes = {};
+
+  // TODO should change this to pulling namedValues from the event instead of reading last line of spreadsheet of responses (the latter is easier for testing, for now)
+  var formResponses = FormApp.getActiveForm().getResponses();
+  var mostRecentResponse = formResponses[formResponses.length - 1];
+  mostRecentResponse.getItemResponses().forEach(function(item) {
+    attributes[item.getItem().getTitle()] = item.getResponse();
+  });
+
+  attributes["sponsor_id"] = getOrCreateSponsor(attributes).id;
+  attributes["show_id"] = getOrCreatePodcast(attributes).id;
+
+  createCampaign(attributes);
+  setChoices(); // reset the answer choices for exising sponsor / podcast by pulling from jingle
+}
+
+function getOrCreateSponsor(attributes) {
+  if (!attributes['Existing sponsor'].match(/\Other/)) {
+    return {id: attributes['Existing sponsor'].split(':')[1].trim()}
+  } else {
+    var newSponsor = {
+      sponsor: {
+        name: attributes['New sponsor name'],
+        billing_info: attributes['New sponsor billing information'],
+        notes: attributes['Notes about new sponsor']
+      }
+    }
+    var sponsorResponse = JSON.parse(postRequest(SPONSOR_URL, newSponsor).getContentText());
+    return {id: sponsorResponse.id};
+  }
+}
+
+function getOrCreatePodcast(attributes) {
+  if (!attributes['Existing podcast'].match(/\Other/)) {
+    return {id: attributes['Existing podcast'].split(':')[1].trim()}
+  } else {
+    var newPodcast = {
+      show: {
+        name: attributes['New podcast name'],
+        network: attributes['New podcast network'],
+        structure: attributes['New podcast show structure'],
+        recording_day: attributes['New podcast recording day of week (if known)'],
+        rate: attributes['New podcast rate'],
+        notes: attributes['Notes about new podcast']
+      }
+    }
+    var podcastResponse = JSON.parse(postRequest(PODCAST_URL, newPodcast).getContentText());
+    return {id: podcastResponse.id};
+  }
+}
+
+function createCampaign(attributes) {
+  var newCampaign = {
+    campaign: {
+      start_date: attributes['Start date'],
+      end_date: attributes['End date'],
+      copy: attributes['Copy'],
+      zone: attributes['Zone(s)'].join(),
+      sponsor_id: attributes['sponsor_id'],
+      show_id: attributes['show_id']
```

Comment:
Is there any case for adding an `approved` flag to the schema for these, or something?  Or maybe not important, since this is just a temporary part of the workflow.

Reply:
not sure what you mean? approved like the sales team has signed off on it?

Reply:
i guess i'm not entirely clear on who fills out this form.  if it's external people, than I was thinking someone internal needed to approve it.

but if it's our internal team, then i suppose we don't need to approve anything.

Reply:
it'll be only our people -- Kali / Madeline.